packages/workbox-runtime-caching/utils/pluginBuilder.mjs

```diff
@@ -0,0 +1,44 @@
+/*
+ Copyright 2016 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import {CacheExpirationPlugin} from
+  'workbox-cache-expiration/CacheExpirationPlugin.mjs';
+import '../_version.mjs';
+
+const pluginBuilder = (options) => {
+  const plugins = [];
+
+  // TODO: Assert plugins is an array
+
+  const pluginParamsToClass = {
+    cacheExpiration: CacheExpirationPlugin,
+    // TODO: Add support for 'broadcastCacheUpdate': BroadcastCacheUpdatePlugin,
+    // TODO: Add support for 'cacheableResponse': CacheableResponsePlugin,
+  };
+
+  // Iterate over known plugins and add them to Request Wrapper options.
```

Comment:
Instead of iterating over known plugins, how about iterating over the `options`, and then logging in `dev` mode when there's an unsupported option used? That sounds more in keeping with our v3 approach to reporting problematic usage that silently ignoring the unsupported config.

Something like:

```js
for (const [pluginName, config] of Object.entries(options)) {
  // Special case for the `cacheName` config.
  if (pluginName === 'cacheName') {
    continue;
  }

  const PluginClass = pluginParamsToClass[pluginName];
  if (PluginClass) {
    plugins.push(new PluginClass(config));
  } else if (process.env.NODE_ENV !== 'production') {
    // Log something about pluginName not being supported.
  }
}
```